apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/GetQueryResultsOptions.cs

```diff
@@ -61,6 +61,13 @@ internal void ModifyRequest(GetQueryResultsRequest request)
             // Nothing to do? We may have something later...
         }
 
+        /// <summary>
```

Comment:
Trying to figure out where this is used, and indeed where options in general are stored - I suspect you know that off the top of your head. Care to help me out? :)

Reply:
this is used to snap the options on the current thread before doing any awaits which could cause a context switch.
Checkout BigQueryConnection.Queries.cs which has a SqlCommandOptions that uses this.
It's eventually to ensure that if a developer calls something like a WaitAll on a linq Select which returns tasks created by ExecuteReader/NonQuery that there arent threading issues.  Code like this:
  Task.WaitAll(  parameters.Select(x => {  
                    bqCommand.Parameter[name]=x; 
                    return bqCommand.ExecuteNonQuery();
                    });
  the options on the command need to be snapped before the first context switch (await) or else you get threading issues with the next cmd invocation.  shouldn't happen often, but I saw it in one of the spanner samples, so I'm just making sure there is consistency here.
Of course, I messed up cmd parameter usage (as in, it isnt working) -- so I need to update the PR again.  fix inc...

Reply:
Ah, I haven't reviewed the generated code yet - that explains it. Will come back to looking at options.